packages/node/src/evm/requests/api-calls.ts

```diff
@@ -1,56 +1,57 @@
 import flatMap from 'lodash/flatMap';
 import * as encoding from '../abi-encoding';
-import * as contracts from '../contracts';
+import { airnodeRrpTopics } from '../contracts';
 import * as events from './events';
 import * as logger from '../../logger';
 import {
   ApiCall,
   ApiCallType,
   ClientRequest,
-  EVMEventLogWithMetadata,
+  EVMEventLog,
+  EVMRequestCreatedLog,
+  EVMRequestFulfilledLog,
   LogsData,
   RequestErrorCode,
   RequestStatus,
 } from '../../types';
 
 function getApiCallType(topic: string): ApiCallType {
-  const { topics } = contracts.AirnodeRrp;
   switch (topic) {
-    case topics.ClientRequestCreated:
+    case airnodeRrpTopics.ClientRequestCreated:
       return 'regular';
-    case topics.ClientFullRequestCreated:
+    case airnodeRrpTopics.ClientFullRequestCreated:
       return 'full';
     // This should never be reached
     default:
       throw new Error(`Unknown topic:${topic} during API call initialization`);
   }
 }
 
-export function initialize(logWithMetadata: EVMEventLogWithMetadata): ClientRequest<ApiCall> {
-  const { parsedLog } = logWithMetadata;
+export function initialize(log: EVMRequestCreatedLog): ClientRequest<ApiCall> {
+  const { parsedLog } = log;
 
   const request: ClientRequest<ApiCall> = {
     airnodeId: parsedLog.args.airnodeId,
     chainId: parsedLog.args.chainId.toString(),
     clientAddress: parsedLog.args.clientAddress,
-    designatedWallet: parsedLog.args.designatedWallet || null,
+    designatedWallet: parsedLog.args.designatedWallet,
     encodedParameters: parsedLog.args.parameters,
     id: parsedLog.args.requestId,
-    endpointId: parsedLog.args.endpointId || null,
-    fulfillAddress: parsedLog.args.fulfillAddress || null,
-    fulfillFunctionId: parsedLog.args.fulfillFunctionId || null,
+    endpointId: events.isFullApiRequest(log) ? log.parsedLog.args.endpointId : null,
```

Comment:
Do we need to check the event type like this?

Reply:
`endpointId` is only defined in full request. This was the cleanest way I found to make this readable and satisfy TS static checkings.